modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java

```diff
@@ -1522,6 +1522,22 @@ public CPDefinition fetchCPDefinitionByCProductId(long cProductId) {
 			cProduct.getCProductId(), cProduct.getLatestVersion());
 	}
 
+	@Override
+	public CPDefinition getCPDefinitionByFriendlyURLPath(
+		long groupId, String friendlyURLPath) {
+
+		FriendlyURLEntry friendlyURLEntry =
+			_friendlyURLEntryLocalService.fetchFriendlyURLEntry(
+				groupId, _classNameLocalService.getClassNameId(CProduct.class),
+				friendlyURLPath);
+
+		if (friendlyURLEntry == null) {
+			return null;
+		}
+
+		return fetchCPDefinitionByCProductId(friendlyURLEntry.getClassPK());
```

Comment:
use this
`return cpDefinitionLocalService.fetchCPDefinitionByCProductId(friendlyURLEntry.getClassPK());`